crates/core/component/stake/src/component/stake.rs

```diff
@@ -487,6 +487,32 @@ pub trait ConsensusIndexRead: StateRead {
             })
             .boxed())
     }
+
+    /// Returns whether the given validator should be indexed in the consensus set.
+    async fn belongs_in_index(&self, validator_id: &IdentityKey) -> bool {
+        let Some(state) = self
+            .get_validator_state(validator_id)
+            .await
+            .expect("no deserialization error")
+        else {
+            tracing::error!(
+                %validator_id,
+                "consenus set index check returned false because validator state was not found");
+            return false;
+        };
+
+        let belongs_in_index =
+            matches!(state, validator::State::Active | validator::State::Inactive);
+
+        tracing::debug!(
+            %validator_id,
+            ?belongs_in_index,
+            ?state,
+            "removing validator from consensus set"
+        );
```

Comment:
i think this might also be slightly inaccurate, we'd only remove the validator if this is `false`, if `belongs_in_index` is true we'd print this log still

Reply:
amended the log string, thx!

Reply:
that commit didnt really address the point, i'll mark this back as unresolved if you don't mind

Reply:
(i don't want to bog this down so i'll offer you a patch)

Reply:
@erwanor a5f86fee3